### PR TITLE
Add Support for Fetching and Deploying DuplicateRules

### PR DESCRIPTION
**Pull request by cwarden**

### CODE REVIEW

**Comment by yurybond** [comment]
Do we need to add suport of matching rules as well https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_matchingrule.htm

File: lib/packagebuilder.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ var metapaths = []metapath{
 	metapath{path: "LeadConvertSettings", name: "LeadConvertSettings"},
 	metapath{path: "letterhead", name: "Letterhead"},
 	metapath{path: "matchingRules", name: "MatchingRules"},
```
Comment (yurybond) [inline lib/packagebuilder.go]: this file already has matchingRule path (line 86) but with wrong name (plural), should be singular. please remove line above

Comment (cwarden) [reply lib/packagebuilder.go]: Take a look at the [examples in the docs](https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_matchingrule.htm).  This metadata object is unusual in that it uses the plural name when fetching using a wildcard:

```
<?xml version="1.0" encoding="UTF-8"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
  <types>
    <members>*</members>
    <name>MatchingRules</name>
  </types>
  <version>42.0</version>
</Package>
```

Comment (yurybond) [reply lib/packagebuilder.go]: so, perhaps, you need to remove the next line (line 87)? 
```
metapath{path: "matchingRules", name: "MatchingRule"},
```

Comment (cwarden) [reply lib/packagebuilder.go]: I think they're both needed to support `force fetch -t MatchingRules` and `force fetch -t MatchingRule -n Account.AccountMatchingRule`.

Have you found any problems with having both?

Comment (yurybond) [reply lib/packagebuilder.go]: I just worried about duplication and incorrect package.xml generation for `force push -f ` `force push -t -n`

I did not test it yet, to be honest

**Pull request merged**